python/cudf/cudf/core/column/column.py

```diff
@@ -1273,6 +1274,32 @@ def scatter_to_table(
             }
         )
 
+    def _apply_type_metadata(self: ColumnBase, dtype: Dtype) -> ColumnBase:
```

Comment:
Can this be used also in the `_copy_type_metadata_from_arrow` function below?

Reply:
I think so, I was a little confused by the recursive call on the children being used as an argument for column construction here:

https://github.com/rapidsai/cudf/blob/69c8373564edb6d9dc792908f9cfc76922d7d02a/python/cudf/cudf/core/column/column.py#L2221-L2234

But I don't think that should conflict.